drivers/storage/portworx/components.go

```diff
@@ -2012,6 +2059,189 @@ func (p *portworx) createPortworxAPIDaemonSet(
 	return k8sutil.CreateOrUpdateDaemonSet(p.k8sClient, newDaemonSet, ownerRef)
 }
 
+func (p *portworx) createServiceMonitor(
+	cluster *corev1alpha1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	svcMonitor := &monitoringv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pxServiceMonitor,
+			Namespace: cluster.Namespace,
+			Labels: map[string]string{
+				"name": pxServiceMonitor,
+			},
+			OwnerReferences: []metav1.OwnerReference{*ownerRef},
+		},
+		Spec: monitoringv1.ServiceMonitorSpec{
+			Endpoints: []monitoringv1.Endpoint{
+				{
+					Port: pxRESTPortName,
+				},
+			},
+			NamespaceSelector: monitoringv1.NamespaceSelector{
+				MatchNames: []string{cluster.Namespace},
+			},
+			Selector: metav1.LabelSelector{
+				MatchLabels: p.GetSelectorLabels(),
+			},
+		},
+	}
+
+	// In case kvdb spec is nil, we will default to internal kvdb
+	if cluster.Spec.Kvdb == nil || cluster.Spec.Kvdb.Internal {
+		svcMonitor.Spec.Endpoints = append(
+			svcMonitor.Spec.Endpoints,
+			monitoringv1.Endpoint{Port: pxKVDBPortName},
+		)
+	}
+
+	return k8sutil.CreateOrUpdateServiceMonitor(p.k8sClient, svcMonitor, ownerRef)
+}
+
+func (p *portworx) createPrometheusRule(
```

Comment:
wouldn't this be very difficult to manage. Whenever we want to add new rules, we will end up modifying at multiple places - here in operator, installer spec (for daemonset) and our docs.

Instead can we just maintain one spec at install.portworx.com. Operator tries to fetch this from yaml from install.portworx.com. If it cannot reach then it will have one spec baked in the container for each version.

Reply:
That's fine we can also just add additional rule objects.